src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -5596,8 +5596,8 @@ private static String replace(final String text, String searchString, final Stri
          }
          String searchText = text;
          if (ignoreCase) {
-             searchText = text.toLowerCase();
-             searchString = searchString.toLowerCase();
+             searchText = text.toUpperCase();
+             searchString = searchString.toUpperCase();
```

Comment:
Just leaving a comment here too to have a review here in GitHub. While your example works, as the character is considered already in upper case, the reverse case would still fail after changing from `toLowerCase` to `toUpperCase`. So I think we should find another solution or update the documentation stating how the code works with unicode.

Reply:
Oohh, great testing @HiuKwok ! Thanks for sharing here.

>Just a quick question what you mean by remove the length( ) mean? Would you mind to specify more on that?

Sure. I think there could be a possibility to fix the issue by addressing how the length of the lower'ed/upper'ed text is used https://github.com/apache/commons-lang/blob/590f90889bf61a5570bd98b78e73410a07d7410b/src/main/java/org/apache/commons/lang3/StringUtils.java#L5603

So maybe there could be another way to work around the way we use the strings lengths, and avoid the exception.

Reply:
@kinow May be try to store a original text.length before performance any upper || lower case operation and make use of it?

haha, it's just a thought. Would try out later

Reply:
@kinow It turn out the problem is large than that, because the incorrect length basically affecting the whole method, especially the part which involve parsing inside the while loop.

https://github.com/apache/commons-lang/blob/590f90889bf61a5570bd98b78e73410a07d7410b/src/main/java/org/apache/commons/lang3/StringUtils.java#L5612-L5619

While string like following is entered, then Exception would be thrown on here instead.
https://github.com/apache/commons-lang/blob/590f90889bf61a5570bd98b78e73410a07d7410b/src/main/java/org/apache/commons/lang3/StringUtils.java#L5613

![image](https://user-images.githubusercontent.com/37996731/44082531-137cb1a0-9fe4-11e8-9219-8f373aebce67.png)

Reply:
Yeah, a very slippery problem. We still have the option to simply document that the method does not work well with unicode strings. 

But first I would like to spend at least a few hours with paper and pencil (and eraser, because this could take a bit till I give up or find a possible way around it), and perhaps even check in the mailing list if other devs have any idea.

I think you found a very interesting problem (*)! Keep the ideas coming if you have any on how to solve this issue :+1: 

_* if I had more time, I would possibly either go through other methods checking for that or, just try some fuzzifier approach to test the whole project ! Not aware of any static or dynamic analysis tool that does that_

Reply:
Hi all, Just a quick update finally I kind of figure the way out of this problem,
while I am trying to make it happen (the implementation).
But the concept is mainly Normalize incoming word at the beginning of the method call by ```Normalizer.normalize()```